src/k2/tso/service/TSOService.h

```diff
@@ -150,17 +149,12 @@ class TSOService::TSOController
     private:
 
     // Design Note:
-    // 1. Interaction between controller and workers
-    //    a) during start(), controller collect workers URLs
-    //       and if controller after JoinServerCluster() find itself is master,
-    //       it will UpdateWorkerControlInfo() through out of band DoHeartBeat() to enable workers start serving requests
-    //    b) Once started, controller will only UpdateWorkerControlInfo() through regular HeartBeat()
-    // 2. Internally inside controller
-    //    a) during start(), initialize controller JoinServerCluster() and if master, update workers through out of band DoHeartBeat()
-    //    b) Once started, only periodically HeartBeat() will handle all complex logic including role change, updating worker, handle gracefully stop()/lost lease suicide().
-    //    c) TimeSyn() only update in memory _controlInfoToSend, which will be sent with next HeartBeat();
+    //    a) during start(), controller collect workers URLs and issue first time DoTimeSync() and update in memory _controlInfoToSend
```

Comment:
I think we should start with a call to CPO to discover current TSOs in the cluster. Then we should attempt to join them.

Reply:
En, as we discussed, since CPO is not dependent on TSO, we can do so and let CPO owns the TSO node list. I will file an issue for this and make change in a different PR.